scripts/lib/libvirt/cleanup.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import libvirt
+import sys
+import os
+import glob
+import subprocess
+import argparse
+
+conn = libvirt.open("qemu:///system")
+if not conn:
+    print "Failed to open connection to the hypervisor"
+    sys.exit(1)
+
+parser = argparse.ArgumentParser(description="Cleanup libvirt resources.")
+parser.add_argument("cloud", type=str, help="Name of the cloud")
+parser.add_argument("nodenumber", type=int, help="Number of nodes")
+parser.add_argument("cloudbr", type=str, help="Name of the virtual bridge")
+parser.add_argument("public_vlan", type=str, help="Name of the public vlan")
+
+args = parser.parse_args()
+
+def remove_files(files):
+    for f in glob.glob(files):
+        print "removing {}".format(f)
+        os.remove(f)
+
+def main():
+    devnull = open(os.devnull, "w")
+    allnodenames = ["admin"]
+    allnodenames.extend("node%s"%i for i in range(1, args.nodenumber+20))
+
+    try:
+        for nodename in allnodenames:
+            vm = "{}-{}".format(args.cloud, nodename)
+            dom = conn.lookupByName(vm)
+            if dom.isActive():
+                print "destroying {}".format(dom.name())
+                dom.destroy()
+            if dom.ID():
+                print "undefining {}".format(dom.name())
+                dom.undefine()
+
+            machine = "{}-{}".format("qemu", vm)
+            machinectl = subprocess.call(["test", "-x", "/usr/bin/machinectl"])
+            machine_status = subprocess.call(["machinectl", "status", machine], stdout=DEVNULL, stderr=subprocess.STDOUT)
+            if (machinectl == 0 and machine_status == 0):
+                subprocess.call(["machinectl", "terminate", machine]) # workaround bnc#916518
+    except Exception:
+        print "...skipping undefined domains"
+
+    net_name = "{}-admin".format(args.cloud)
+    networks = [n for n in conn.listAllNetworks if n.name() == net_name]
```

Comment:
You can simply use `conn.networkLookupByName(net_name)`. Then you don't need the loops afterwards